examples/text_to_image/train_text_to_image_sdxl.py

```diff
@@ -0,0 +1,1206 @@
+            "The name of the Dataset (from the HuggingFace hub) to train on (could be your own, possibly private,"
+            " dataset). It can also be a path pointing to a local copy of a dataset in your filesystem,"
+            " or to a folder containing files that 🤗 Datasets can understand."
+        ),
+    )
+    parser.add_argument(
+        "--dataset_config_name",
+        type=str,
+        default=None,
+        help="The config of the Dataset, leave as None if there's only one config.",
+    )
+    parser.add_argument(
+        "--train_data_dir",
+        type=str,
+        default=None,
+        help=(
+            "A folder containing the training data. Folder contents must follow the structure described in"
+            " https://huggingface.co/docs/datasets/image_dataset#imagefolder. In particular, a `metadata.jsonl` file"
+            " must exist to provide the captions for the images. Ignored if `dataset_name` is specified."
+        ),
+    )
+    # TODO: Fix this
+    # parser.add_argument(
+    #     "--original_image_column",
+    #     type=str,
+    #     default="input_image",
+    #     help="The column of the dataset containing the original image on which edits where made.",
+    # )
+    # parser.add_argument(
+    #     "--edited_image_column",
+    #     type=str,
+    #     default="edited_image",
+    #     help="The column of the dataset containing the edited image.",
+    # )
+    # parser.add_argument(
+    #     "--edit_prompt_column",
+    #     type=str,
+    #     default="edit_prompt",
+    #     help="The column of the dataset containing the edit instruction.",
+    # )
+    
+    # TODO: Delete this.
+    # parser.add_argument(
+    #     "--val_image_url_or_path",
+    #     type=str,
+    #     default=None,
+    #     help="URL to the original image that you would like to edit (used during inference for debugging purposes).",
+    # )
+    parser.add_argument(
+        "--validation_prompt", type=str, default=None, help="A prompt that is sampled during training for inference."
+    )
+    parser.add_argument(
+        "--num_validation_images",
+        type=int,
+        default=4,
+        help="Number of images that should be generated during validation with `validation_prompt`.",
+    )
+    parser.add_argument(
+        "--validation_steps",
+        type=int,
+        default=100,
+        help=(
+            "Run fine-tuning validation every X steps. The validation process consists of running the prompt"
+            " `args.validation_prompt` multiple times: `args.num_validation_images`."
+        ),
+    )
+    parser.add_argument(
+        "--max_train_samples",
+        type=int,
+        default=None,
+        help=(
+            "For debugging purposes or quicker training, truncate the number of training examples to this "
+            "value if set."
+        ),
+    )
+    parser.add_argument(
+        "--output_dir",
+        type=str,
+        default="instruct-pix2pix-model",
```

Comment:
=> `pipeline`

Reply:
It's not an InstructPix2Pix pipeline. We can call it `"text-to-image-sdxl"`.

Reply:
done in next commit